lib/signaling/v2/peerconnection.js

```diff
@@ -276,6 +277,21 @@ class PeerConnectionV2 extends StateMachine {
     return this._peerConnection.iceConnectionState;
   }
 
+  /**
+   * Whether the {@link PeerConnectionV2} has negotiated or is in the process
+   * of negotiating the application m= section.
+   * @returns {boolean}
+   */
+  get isApplicationSectionNegotiated() {
+    // NOTE(mmalavalli): Because of VIDEO-1981, we only avoid re-negotiations
+    // for LocalDataTrack changes when the remote peer is an SFU.
+    // TODO(mmalavalli): Remove the ICE-lite agent check once VIDEO-1981
+    // is fixed.
+    return this._isIceLite && this._peerConnection.localDescription
+      ? getMediaSections(this._peerConnection.localDescription.sdp, 'application').length > 0
+      : false;
+  }
```

Comment:
We have a similar check for the presence of an `m=application` section in C++, so that we can re-negotiate to add the section if needed. 

I am slightly confused about the VIDEO-1981 workaround, because the ticket describes Peer-to-Peer and your comment mentions SFU.

Reply:
@ceaglest Because of VIDEO-1981, we always need to re-negotiate when we are in a Peer-to-Peer Room. So, we only avoid re-negotiation when we are in a group Room (the assumption here is that an ICE-lite agent is the SFU). Does that clarify things for you?